tests/features/step_definitions/10213.ACs.feature.steps.intTest.ts

```diff
@@ -0,0 +1,133 @@
+import {defineFeature, loadFeature} from "jest-cucumber";
+import supertest from "supertest";
+import path from "path";
+import mockData from "../../resources/technical-records.json";
+import mockContext from "aws-lambda-mock-context";
+import {emptyDatabase, populateDatabase} from "../../util/dbOperations";
+import {cloneDeep} from "lodash";
+
+const url = "http://localhost:3005/";
+const request = supertest(url);
+const opts = Object.assign({
+  timeout: 1.5
+});
+
+const createPOSTPayload = () => {
+  const techRec: any = cloneDeep(mockData[43]);
+  delete techRec.techRecord[0].statusCode;
+  const payload = {
+    msUserDetails: {
+      msUser: "dorel",
+      msOid: "1234545"
+    },
+    vin: Date.now().toString(),
+    techRecord: techRec.techRecord
+  };
+  return payload;
+};
+
+const feature = loadFeature(path.resolve(__dirname, "../10213.ACs.feature"));
+
+defineFeature(feature, (test) => {
+  beforeAll(async () => {
+    await emptyDatabase();
+  });
+
+  beforeEach(async () => {
+    await populateDatabase();
+  });
+
+  afterEach(async () => {
+    await emptyDatabase();
+  });
+
+  afterAll(async () => {
+    await populateDatabase();
+  });
+
+  test("AC1. POST: Partial VIN is autopopulated", ({given, when, then, and}) => {
+    let ctx: any = mockContext(opts);
+
+    let requestUrlPOST: string;
+    let requestUrlGET: string;
+    const postPayload = createPOSTPayload();
+    let responsePOST: any;
+    let responseGET: any;
+    const partialVin: string = postPayload.vin.substr(postPayload.vin.length - 6);
+
+    given("I am a consumer of the vehicles API", () => {
+      requestUrlPOST = "vehicles";
+      requestUrlGET = `vehicles/${partialVin}/tech-records`;
+    });
+    and("I have completed the \"vin\" field", () => {
+      expect(postPayload.vin).toBeDefined();
+    });
+    when("I submit my request via the POST method", async () => {
+      responsePOST = await request.post(requestUrlPOST).send(postPayload);
+    });
+    then("the partialVin is autopopulated, as the last 6 digits of the vin", async () => {
+      responseGET = await request.get(requestUrlGET);
+      expect(responseGET.body.vin).toEqual(postPayload.vin);
```

Comment:
The only way I can test if the partialVin was populated correctly is to do a GET using the partialVin, which I'm doing now, because when creating a new record you don't get back the created record, you only get a message saying "Technical record created". And if I do a GET using the VIN I don't get back the partialVin, because it is deleted. Line 104 TechRecordsService
    // Cleaning up unneeded properties
    delete techRecordItem.primaryVrm; // No longer needed
    delete techRecordItem.secondaryVrms; // No longer needed
    delete techRecordItem.partialVin; // No longer needed